modern/tests/test_build/test_gfortran.py

```diff
@@ -0,0 +1,56 @@
+import pytest
+from fixtures.expected import get_compiler_versions, Version
+
+
+# Checks that a very simple Fortran project can be built using these images
+# It also checks the libraries the project was linked with and that
+# the binary generated can run in a vanilla image and _newer_ images
+
+
+@pytest.fixture(autouse=True, scope='class')
+def _build_simple(request, container, expected):
+    build_directory = 'modern/tests/tmp/gfortran'
+    request.cls._build_directory = build_directory
+
+    with container.working_dir(build_directory):
+        container.exec(['cmake', '../../workingdir/gfortran', '-DCMAKE_BUILD_TYPE=Release'])
+        container.exec(['cmake', '--build', '.'])
+
+
+@pytest.mark.compiler('gcc')
+class TestBuildGFortran:
+
+    @pytest.mark.service('deploy', 'jenkins')
+    def test_running_container(self, container, expected):
+        # Compile the project
+        with container.working_dir(self._build_directory):
+            # Check it runs
+            out, err = container.exec(['./hello'])
+            assert 'Hello world!' in out, f"out: '{out}' err: '{err}'"
+
+            # Check linked libs
+            out, err = container.exec(['ldd', 'hello'])
+            # TODO: Test linked libraries here (gcc)
+            print(out)
+            print(err)
+
+    @pytest.mark.service('deploy', 'jenkins')
+    def test_vanilla_image(self, container, expected):
+        # C executable should run in vanilla image
+        with container.run_container(expected.vanilla_image()) as vanilla:
+            with vanilla.working_dir(self._build_directory):
+                out, err = vanilla.exec(['./hello'])
+                assert 'Hello world!' in out, f"out: '{out}' err: '{err}'"
```

Comment:
```suggestion
        out, err = container.exec(['ls', '-Ral', '/usr/local'])
        container.exec(['cp', '/usr/local/lib64/libgfortran.so.3.0.0', self._build_directory])
        container.exec(['cp', '/usr/local/lib64/libquadmath.so.0.0.0', self._build_directory])
        with container.run_container(expected.vanilla_image()) as vanilla:
            with vanilla.working_dir(self._build_directory):
                vanilla.exec(['cp', 'libgfortran.so.3.0.0', '/usr/lib/libgfortran.so.3'])
                vanilla.exec(['cp', 'libquadmath.so.0.0.0', '/usr/lib/libquadmath.so.0'])
                out, err = vanilla.exec(['./hello'])
                assert 'Hello world!' in out, f"out: '{out}' err: '{err}'"
```

We need to copy libgfortran and libquadmath from container. This suggestion is hardcoded and works for GCC5, but I guess the library version is not the same for gcc11 for instance. Still, we can use wildcard for copying.

Can we solve it using apt-get only?
No, because libgfortran is under gcc apt package
Yes, because we can download any .deb package and extract its content.

Reply:
Ok, if this doesn't work in a _vanilla_ image, probably we shouldn't run this test.